src/lib/eina/eina_inline_thread_posix.x

```diff
@@ -100,14 +100,27 @@ _eina_thread_join(Eina_Thread t)
 }
 
 static inline Eina_Bool
-_eina_thread_name_set(Eina_Thread thread, char *buf)
+_eina_thread_name_set(Eina_Thread thread, char *name)
 {
-#ifndef __linux__
-   pthread_set_name_np((pthread_t)t, buf);
+#ifdef EINA_HAVE_PTHREAD_SETNAME
+   char buf[16];
+   if (name)
+     {
+        strncpy(buf, name, 15);
+        buf[15] = 0;
+     }
+   else buf[0] = 0;
+ #ifndef __linux__
+   pthread_set_name_np((pthread_t)thread, buf);
```

Comment:
Shouldn't it be like this?
```suggestion
   pthread_setname_np((pthread_t)thread, buf);
```

Reply:
So this is o an `#ifndef __linux__` branch, I think it uses the [FreeBSD functions](https://www.freebsd.org/cgi/man.cgi?query=pthread_set_name_np&sektion=3&apropos=0&manpath=FreeBSD+7.1-RELEASE).

Reply:
Ok, got it.